homeassistant/components/sense/sensor.py

```diff
@@ -291,6 +316,29 @@ def native_value(self):
         """Return the state of the sensor."""
         return round(self._data.get_trend(self._sensor_type, self._variant_id), 1)
 
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+
+        if self._sensor_type == "DAY":
+            new_update = dt_util.now().day
+        elif self._sensor_type == "WEEK":
+            new_update = dt_util.now().day
+        elif self._sensor_type == "MONTH":
+            new_update = dt_util.now().month
+        elif self._sensor_type == "YEAR":
+            new_update = dt_util.now().year
+        else:
+            new_update = -1
+        if self._last_update != new_update:
```

Comment:
This logic will make the weekly sensor change its `last_reset` every day, not once per week.
Is that the intention?

Reply:
~Yes, the weekly sensor is a bit different from the rest as it's a rolling window of the previous 7 days, not a calendar week~
Just double checked and it IS returning the calendar week so should be reset on Monday